pennylane/ops/qubit.py

```diff
@@ -483,13 +569,18 @@ class RZ(Operation):
     grad_method = "A"
     generator = [PauliZ, -1 / 2]
 
-    @staticmethod
-    def _matrix(*params):
+    @classmethod
+    def _matrix(cls, *params):
         theta = params[0]
-        return np.cos(theta / 2) * np.eye(2) + 1j * np.sin(-theta / 2) * PauliZ._matrix()
+        return np.array([[cmath.exp(-0.5j * theta), 0], [0, cmath.exp(0.5j * theta)]])
```

Comment:
Just a thought, could using `conj` be slightly faster?

Reply:
Probably,I'll have a look at the timing

Reply:
It is
```
In [6]: %timeit p = cmath.exp(0.5j * theta); p.conjugate()
306 ns ± 26.1 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [7]: %timeit cmath.exp(0.5j * theta), cmath.exp(-0.5j * theta)
493 ns ± 16.5 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
hmm, not insignificant either